client/cmd/dexc-desktop/syncserver.go

```diff
@@ -0,0 +1,163 @@
+// This code is available on the terms of the project LICENSE.md file,
+// also available online at https://blueoakcouncil.org/license/1.0.0.
+
+/*
+The sync server is a tiny HTTP server that handles synchronization of multiple
+instances of dexc-desktop. Any dexc-desktop instance that manages to get a
+running webserver will create a sync server and publish the address at specific
+file location. On startup, before getting too far, dexc-desktop looks for a
+file, reads the address, and attempts to make contact. If contact is made,
+the requesting dexc-desktop will exit without error. The receiving dexc-desktop
+will reopen the window if it is closed.
+
+The sync server also enables killing a dexc-desktop process that is running
+in the background because of active orders.
+*/
+
+package main
+
+import (
+	"context"
+	"errors"
+	"net"
+	"net/http"
+	"os"
+	"path/filepath"
+	"sync"
+	"time"
+
+	"decred.org/dcrdex/client/app"
+	"decred.org/dcrdex/dex"
+)
+
+const (
+	syncFilename = "syncfile"
+)
+
+// getSyncAddress: Get the address stored in the syncfile. If the file doesn't
+// exist, an empty string is returned with no error.
+func getSyncAddress(syncDir string) (addr string, err error) {
+	syncFile := filepath.Join(syncDir, syncFilename)
+	addrB, err := os.ReadFile(syncFile)
+	if err != nil {
+		if !os.IsNotExist(err) {
+			return "", err
+		}
+		return "", nil
+	}
+	return string(addrB), nil
+}
+
+// sendKillSignal yeets a kill command to any address in the sync file.
+func sendKillSignal(syncDir string) {
+	addr, err := getSyncAddress(syncDir)
+	if err != nil {
+		log.Errorf("Error getting sync address for kill signal: %v", err)
+		return
+	}
+	if addr == "" {
+		log.Errorf("No sync address found for kill signal: %v", err)
+		return
+	}
+	resp, err := http.Get("http://" + addr + "/kill")
+	if err != nil {
+		log.Errorf("Error sending kill signal to sync server: %v", err)
+		return
+	}
+	if resp == nil {
+		log.Errorf("nil kill response")
+		return
+	}
+	if resp.StatusCode != http.StatusOK {
+		log.Errorf("Unexpected response code from kill signal send: %d (%s)", resp.StatusCode, http.StatusText(resp.StatusCode))
+	}
+}
+
+// synchronize attempts to determine the state of any other dexc-desktop
+// instances that are running. If none are found, startServer will be true.
+// If another instance is found, close will be true, and this instance of
+// dexc-desktop should exit immediately.
+func synchronize(syncDir string) (startServer, close bool, err error) {
```

Comment:
When `err == nil`, the bools predict each other.  I think there should be a single bool.